filters/auth/tokeninfo_test.go

```diff
@@ -15,176 +17,174 @@ import (
+		args:           []interface{}{testRealmKey, testRealm, testKey, testValue},
+		auth:           testToken,
+		expected:       http.StatusOK,
+		expectRequests: N,
 	}, {
-		msg:         "allKV(): valid token, one valid key value pair",
-		authType:    filters.OAuthTokeninfoAllKVName,
-		authBaseURL: testAuthPath,
-		args:        []interface{}{testKey, testValue},
-		hasAuth:     true,
-		auth:        testToken,
-		expected:    http.StatusOK,
+		msg:            "allKV(): valid token, valid key value pairs",
+		authType:       filters.OAuthTokeninfoAllKVName,
+		args:           []interface{}{testKey, testValue, testKey, testValue},
+		auth:           testToken,
+		expected:       http.StatusOK,
+		expectRequests: N,
 	}, {
-		msg:         "allKV(): valid token, one valid key value pair, check realm",
-		authType:    filters.OAuthTokeninfoAllKVName,
-		authBaseURL: testAuthPath,
-		args:        []interface{}{testRealmKey, testRealm, testKey, testValue},
-		hasAuth:     true,
-		auth:        testToken,
-		expected:    http.StatusOK,
+		msg:            "allKV(): valid token, one valid kv, multiple key value pairs1",
+		authType:       filters.OAuthTokeninfoAllKVName,
+		args:           []interface{}{testKey, testValue, "wrongKey", "wrongValue"},
+		auth:           testToken,
+		expected:       http.StatusForbidden,
+		expectRequests: N,
 	}, {
-		msg:         "allKV(): valid token, valid key value pairs",
-		authType:    filters.OAuthTokeninfoAllKVName,
-		authBaseURL: testAuthPath,
-		args:        []interface{}{testKey, testValue, testKey, testValue},
-		hasAuth:     true,
-		auth:        testToken,
-		expected:    http.StatusOK,
+		msg:            "allKV(): valid token, one valid kv, multiple key value pairs2",
+		authType:       filters.OAuthTokeninfoAllKVName,
+		args:           []interface{}{"wrongKey", "wrongValue", testKey, testValue},
+		auth:           testToken,
+		expected:       http.StatusForbidden,
+		expectRequests: N,
 	}, {
-		msg:         "allKV(): valid token, one valid kv, multiple key value pairs1",
-		authType:    filters.OAuthTokeninfoAllKVName,
-		authBaseURL: testAuthPath,
-		args:        []interface{}{testKey, testValue, "wrongKey", "wrongValue"},
-		hasAuth:     true,
-		auth:        testToken,
-		expected:    http.StatusForbidden,
-	}, {
-		msg:         "allKV(): valid token, one valid kv, multiple key value pairs2",
-		authType:    filters.OAuthTokeninfoAllKVName,
-		authBaseURL: testAuthPath,
-		args:        []interface{}{"wrongKey", "wrongValue", testKey, testValue},
-		hasAuth:     true,
-		auth:        testToken,
-		expected:    http.StatusForbidden,
+		msg:            "oauthTokeninfoAnyScope: valid token, one valid scope, cache enabled",
+		authType:       filters.OAuthTokeninfoAnyScopeName,
+		options:        TokeninfoOptions{CacheSize: 1},
+		args:           []interface{}{testScope},
+		auth:           testToken,
+		expected:       http.StatusOK,
+		expectRequests: 1,
```

Comment:
Please add a test case what happens on error cases with enabled Cache.
We don't want to cache invalid tokens (otherwise someone can send us huge token values and we run out of mmemory).

Reply:
Makes sense, added a test to ensure we don't cache invalid token.